### DISCUSSION TITLE
How to set a permanent/static container name?

### DISCUSSION THREAD

**Discussion by dreaded369** [caprover/caprover#1968] [Q&A] [answered]
I'm setting up Volumizer to backup my volumes, and it allows you to specify containers which should be stopped and started around the backup process.
CapRover generates a unique container name each time settings are updated.

Is there a way to override this behaviour and assign a static name to a container?

**Comment by githubsaturn** [comment]
CapRover doesn't do this. It's docker swarm and that's by design.

The static names are service names, you can view them by running `docker service ls`

**Reply by dreaded369** [reply]
Thanks! I'm learning a tonne about Docker using CapRover.

**Reply by dreaded369** [reply]
I guess I'm stuck then - trying to use the static name from 'docker service ls' doesn't work for targeting the container - not using exec or passing it to volumizer as a container to interact with.

**Reply by dreaded369** [reply, marked as answer]
As usual I'm not the first to stumble across an issue: https://github.com/blacklabelops/volumerize/issues/2

For others stumbling across this and exploring backup options:
Volumerize can't handle the hash that docker swarm introdues to container names, so it cannot stop and start containers before and after a backup.
https://github.com/futurice/docker-volume-backup uses labels for stopping and starting containers before/after backups, and since CapRover doesn't support labels it won't work.

**Reply by adamshand** [reply]
@dreadedhamish thanks for this post, you answers a couple of my questions I hadn't dug into yet.  

I believe that CapRover does support lables via service overrides.  Have a look here for examples:

https://github.com/caprover/caprover/issues/38

**Reply by dreaded369** [reply]
I'll look into it. My next step was going to be installing cronicle independently of CapRover (and using the CapRover command to start/stop all), but I'd prefer to keep everything nicely within CapRover.
